src/plugins/intel_gpu/src/plugin/variable_state.cpp

```diff
@@ -81,11 +82,6 @@ void VariableState::set_state(const ov::SoPtr<ov::ITensor>& state) {
             size_t padded_size = src_stride[i] / src_stride[i + 1];
             size_t non_padded_size = src_stride_no_pad[i] / src_stride_no_pad[i + 1];
             int32_t pad_dim_legacy = i + 1;
-            if (pad_dim_legacy >= 2) {
```

Comment:
I think there's no need to extend pad rank anymore on lines 75-76

Reply:
done